### PR TITLE
[fixes #383] ensure every Exception inherits from KazooException

### LINKED ISSUE #383: KazooTimeoutError does not inherit from KazooException
Is there a specific reason for `KazooTimeoutError` inheriting directly from `Exception` instead of from `KazooException`?

### PR DESCRIPTION
**Pull request by Idem**
Fixing #383

### CODE REVIEW

File: kazoo/handlers/eventlet.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 from eventlet.green import threading as green_threading
 from eventlet import queue as green_queue
 
+from kazoo.exceptions import KazooTimeoutError, TimeoutError
```
Comment (Idem) [inline kazoo/handlers/eventlet.py]: As per https://github.com/python-zk/kazoo/pull/402#issuecomment-244782581 the goal is too allow one to keep importing TimeoutError from this module for compatibility.
If that is ok for you @bbangert , I can add an ignore mark on that error.

Comment (bbangert) [inline kazoo/handlers/eventlet.py]: I think that's fine, as it keep's the API consistent then. You can add ``# noqa`` at the end of the line and it'll pass the flake8.

**Comment by bbangert** [comment]
@Idem Did you want to update this so I can merge it?

**Comment by Idem** [comment]
I was looking into the contributing guidelines and where into the process of signing the commits and co.

I obviously need to add the #noqa for hound and pass the travis-ci.
So yes, I intend to clean this up a little more.

**Comment by bbangert** [comment]
I triggered the build by updating base just to ensure all the tests look good still.

File: kazoo/handlers/eventlet.py (hunk 1 above)
Comment (tonyseek) [inline kazoo/handlers/eventlet.py]: Another way to suppress the lint is adding a `__all__ = [..., 'TimeoutError']` line.

**Comment by bbangert** [comment]
Just a heads-up, we're coming up on 30 days now. I'm trying to avoid having PR's idle eternally in the backlog like this, so if you'd like to try later, a new PR can be opened when ready. This is a fairly basic change, so I could clean it up and commit with the appropriate commit message if you'd like.

**Comment by bbangert** [comment]
@Idem Hi! I'd be happy to help with fixing the commit if you need any assistance. A good commit message for this would look like:
```
fix: ensure every exception inherit from KazooException

Closes #383
```

You could rebase this branch against master, amend the commit, then force-push to update the PR.

**Comment by bbangert** [comment]
Closing this for now, please re-open if/when you have time to update it.

**Pull request closed without merging**